### PR TITLE
adding sidecarEnvJson annotation

### PR DESCRIPTION
**Pull request by dileepng**
*Issue #, if available:*
https://github.com/aws/aws-app-mesh-controller-for-k8s/issues/724
*Description of changes:*
Adding Feature appmesh.k8s.aws/sidecarEnvJson that accepts json list as env for envoy.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/inject/envoy.go [hunk 1]
```diff
@@ -249,6 +259,45 @@ func (m *envoyMutator) getCustomEnv(pod *corev1.Pod) (map[string]string, error)
 	return customEnv, nil
 }
 
+type customEnvJsonType []map[string]string
+
+func (c *customEnvJsonType) UnmarshalJSON(data []byte) error {
```
Comment (ysdongAmazon) [inline pkg/inject/envoy.go:264]: Need we make a valid check for bytes before unmarshal if the input is empty or nil?

Comment (dileepng) [reply pkg/inject/envoy.go:264]: It will throw malformed error. so, I didn't make a check for that condition.

Comment (ysdongAmazon) [reply pkg/inject/envoy.go:264]: Thanks. Good to know

File: pkg/inject/envoy.go [hunk 2]
```diff
@@ -249,6 +259,45 @@ func (m *envoyMutator) getCustomEnv(pod *corev1.Pod) (map[string]string, error)
 	return customEnv, nil
 }
 
+type customEnvJsonType []map[string]string
```
Comment (joesbigidea) [inline pkg/inject/envoy.go:262]: why are you not just using a map[string]string?

File: pkg/inject/envoy.go [hunk 3]
```diff
@@ -249,6 +259,41 @@ func (m *envoyMutator) getCustomEnv(pod *corev1.Pod) (map[string]string, error)
 	return customEnv, nil
 }
 
+type customEnvJsonType map[string]string
```
Comment (BennettJames) [inline pkg/inject/envoy.go:262]: Does this custom type still need to exist, or can `getCustomEnvJson` just deserialize to `map[string]string` directly?

Comment (BennettJames) [reply pkg/inject/envoy.go:262]: The only place I see this type used is in the `getCustomEnvJson` function, which immediately unwraps it to a `map[string]string`.

Generally, you only need to some serialization-time changes between the representative data type and the target data type (for instance, when converting strings to dates). I think that might have been true of the first draft of this where the input data was a list of strings, but now that it's just a map of string->string it's the same as what the target type is. Just passing `map[string]string` to unmarshal should accomplish this fine without the indirection.

**Review by BennettJames — APPROVED** [review]

**Pull request merged**